### ISSUE TITLE
Peer Review

### ISSUE THREAD

**Issue by XhoanaShkajoti** [poliSimoneScalora/CI2024_lab3#1]
### *A * with Manhattan distance and other heuristics*
First off I really like your solution, your ideas according the heuristics line up with mine so I am bound to appreciate the solution 😁 . 
### **Strength:**
1. The heuristics: the  combination of Manhattan Distance, the sum of the vertical and horizontal distances of each tile from its goal position, and Linear Conflict, additional penalties for tiles that are in the same row or column but in the wrong order offer for a great optimization in directing the A * into the right answer faster ,enhancing accuracy of simple Manhattan distance and reduces the solution space.
2. Well-Structured Functions:  all your functions are reusable giving a well structured code. 
3. Detailed Outputs: Provides comprehensive outputs, including:
Initial state.
Quality (steps to solution).
Cost (actions evaluated).
Efficiency (quality-to-cost ratio).
Final goal state.
4. is solved function:  Gives the possibility to enter matrixes without randomization although if we don't plan to enter matrixes that don' t come from randomization we don't need it , since its solvable for sure and we save time and resources.

### **Suggestions for Improvement**
1. Documentation and Comments: I would suggest adding more comments to explain the flow logic of the code. Also a table showcasing the code efficiency for changing dimensions would be very helpful since running for N >4 takes a lot of time.
2. Edge Case Handling: Explain and handle edge cases where no solution might be found due to constraints (e.g., time or memory limits).  for me for example beyond N 8 my laptop ram cannot handle it. 
3. Scalability: Document the limitations of the current implementation for puzzles with dimensions 
n>4, as the computational complexity grows rapidly. I would recommend using multiplication factors for scaling dimensions as to overestimate for larger N, also weighted heuristics  can be helpful in increasing efficiency.

### **In summary:** 
 This was a well thought solution that combines what we have learned with done researched and study of the solution. Well done!
